### ISSUE TITLE
Support latest nx version

### ISSUE THREAD

**Issue by jbojcic1** [flowaccount/nx-plugins#72]
**Is your feature request related to a problem? Please describe.**
Atm the plugin has nx "^9.2.2" as a peer dep. Latest nx version is 11 

**Describe the solution you'd like**
Upgrade to latest version

**Comment by jbojcic1** [comment]
I am willing to help with this if needed

**Comment by wickstargazer** [comment]
Hi@ @jbojcic1  i have a branch thats migrated but there are bugs and tests that needs to be written.
Atm i am trying to focus on version 0.5.3 to make the experience better and less buggy then move to migration
would you like to help on that?

**Comment by jbojcic1** [comment]
Sure. Let me know what exactly do you need help with and I'll see what I can do

**Comment by Nightbr** [comment]
Hey, I'm using Nx 11 and try to make `@flowaccount/nx-serverless` works with it.

Here is the Nx report:
```
>  NX  Report complete - copy this into the issue template

  Node : 12.20.1
  OS   : linux x64
  yarn : 1.22.5
  
  nx : Not Found
  @nrwl/angular : Not Found
  @nrwl/cli : 11.2.10
  @nrwl/cypress : 11.2.10
  @nrwl/devkit : 11.2.10
  @nrwl/eslint-plugin-nx : 11.2.10
  @nrwl/express : 9.2.2
  @nrwl/jest : 11.2.10
  @nrwl/linter : 11.2.10
  @nrwl/nest : 11.2.10
  @nrwl/next : Not Found
  @nrwl/node : 11.2.10
  @nrwl/react : 11.2.10
  @nrwl/schematics : Not Found
  @nrwl/tao : 11.2.10
  @nrwl/web : 11.2.10
  @nrwl/workspace : 11.2.10
  typescript : 4.1.3
```

I have installed `yarn add -D  @flowaccount/nx-serverless` which has installed `0.5.2`.

I can generate a lambda function with `nx g @flowaccount/nx-serverless:api-serverless --name=dummy --provider=aws`

But when I want to nx serve lambda-dummy or deploy or another command, it throws this error:

```
Cannot find module '@nrwl/workspace/src/utils/fileutils'
Require stack:
- /home/PATH/node_modules/@flowaccount/nx-serverless/src/utils/serverless.config.js
- /home/PATH/node_modules/@flowaccount/nx-serverless/src/builders/build/build.impl.js
- /home/PATH/node_modules/@nrwl/tao/node_modules/@angular-devkit/architect/node/node-modules-architect-host.js
- /home/PATH/node_modules/@nrwl/tao/node_modules/@angular-devkit/architect/node/index.js
- /home/PATH/node_modules/@nrwl/tao/src/commands/ngcli-adapter.js
- /home/PATH/node_modules/@nrwl/tao/src/commands/run.js
- /home/PATH/node_modules/@nrwl/tao/index.js
- /home/PATH/node_modules/@nrwl/cli/lib/run-cli.js
```

Ready to help & test if someone is working on this :ok_hand:  Let me know how I can help.

**Comment by mandarini** [comment]
Hi there @Nightbr. Thanks for pointing out the Nx-related issue. I only came here to tell that upgrading to `11.3.x` fixes the `utils/utilities` discrepancy, but I see you already found this fix! :) I'll be watching this issue in case some other Nx bug comes up.

**Comment by cpwingert** [comment]
Is there any update / workaround for this issue?  I am running into the same problem.

**Comment by Sam-Dietrich** [comment]
Likewise also running into this issue

**Comment by icedevera** [comment]
Running into this issue as well

**Comment by wickstargazer** [comment]
Hey guys, we will be upgrading the nx-plugin workspace and dependencies to nx 11. But it will take us some time. We welcome collaborators and i do have a branch to begin with as well

**Comment by goldsam** [comment]
Can you push your branch? I might have a little free time in the next week to help with this.

**Comment by wickstargazer** [comment]
hi @goldsam  @jbojcic1  lets get this working, i am going to refactor it out. we can take it out in the current master make it work with 11 then migrate. What do you think?

**Comment by wickstargazer** [comment]
@goldsam by the way this is the branch although the name is obscure `hotfix/make-nx-serverless-run-with-npm`

**Comment by jbojcic1** [comment]
@wickstargazer unfortunately I don't have time to help on this anymore. Will ping you if that changes

**Comment by wickstargazer** [comment]
@jbojcic1 no worries i have been digging around, seems like i have to remove all imports from workspace ( i built the plugin during when there was no devkit) and the generators has changed their structure. So lets see how that goes...probably will need help testing it out to make sure i dont break it.

**Comment by dbrody** [comment]
Any update on this thread? I'm curious to use this with a new NX project.

**Comment by wickstargazer** [comment]
@dbrody i refactored the generators over the weekend. Its building now but i have to migrate the workspace to version that has some new commands.
i am trying to do that atm. But i think we can do a pre-release to test run it first. i will push to the new branch at the end of the day. You are welcome to review the PR together

**Comment by wickstargazer** [comment]
hey guys,
i am still stuck with this error when compiling the nx-serverless. But as a good sign its just some rxjs version change of sytanx error.

anyone has an idea ?

```bash
Compiling TypeScript files for project "nx-serverless"...
libs/nx-serverless/src/utils/target.schedulers.ts:21:9 - error TS2345: Argument of type 'MonoTypeOperatorFunction<BuilderOutput>' is not assignable to parameter of type 'OperatorFunction<BuilderOutput, BuilderOutput>'.
  Types of parameters 'source' and 'source' are incompatible.
    Type 'import("D:/Projects/opensource/flowaccount-nx-plugins/node_modules/rxjs/internal/Observable").Observable<import("D:/Projects/opensource/flowaccount-nx-plugins/node_modules/@angular-devkit/architect/src/api").BuilderOutput>' is not assignable to type 'import("D:/Projects/opensource/flowaccount-nx-plugins/libs/nx-serverless/node_modules/rxjs/internal/Observable").Observable<import("D:/Projects/opensource/flowaccount-nx-plugins/node_modules/@angular-devkit/architect/src/api").BuilderOutput>'.
      The types of 'source.operator.call' are incompatible between these types.
        Type '(subscriber: import("D:/Projects/opensource/flowaccount-nx-plugins/node_modules/rxjs/internal/Subscriber").Subscriber<any>, source: any) => import("D:/Projects/opensource/flowaccount-nx-plugins/node_modules/rxjs/internal/types").TeardownLogic' is not assignable to type '(subscriber: import("D:/Projects/opensource/flowaccount-nx-plugins/libs/nx-serverless/node_modules/rxjs/internal/Subscriber").Subscriber<any>, source: any) => import("D:/Projects/opensource/flowaccount-nx-plugins/libs/nx-serverless/node_modules/rxjs/internal/types").TeardownLogic'.
          Types of parameters 'subscriber' and 'subscriber' are incompatible.
            Type 'import("D:/Projects/opensource/flowaccount-nx-plugins/libs/nx-serverless/node_modules/rxjs/internal/Subscriber").Subscriber<any>' is not assignable to type 'import("D:/Projects/opensource/flowaccount-nx-plugins/node_modules/rxjs/internal/Subscriber").Subscriber<any>'.
              Property 'isStopped' is protected but type 'Subscriber<T>' is not a class derived from 'Subscriber<T>'.

21         filter(e => e.success !== undefined),
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
libs/nx-serverless/src/utils/testing.ts:38:50 - error TS2345: Argument of type 'CoreSchemaRegistry' is not assignable to parameter of type 'SchemaRegistry'.
  The types of 'compile(...).source.operator.call' are incompatible between these types.
    Type '(subscriber: import("D:/Projects/opensource/flowaccount-nx-plugins/libs/nx-serverless/node_modules/rxjs/internal/Subscriber").Subscriber<any>, source: any) => import("D:/Projects/opensource/flowaccount-nx-plugins/libs/nx-serverless/node_modules/rxjs/internal/types").TeardownLogic' is not assignable to type '(subscriber: import("D:/Projects/opensource/flowaccount-nx-plugins/node_modules/rxjs/internal/Subscriber").Subscriber<any>, source: any) => import("D:/Projects/opensource/flowaccount-nx-plugins/node_modules/rxjs/internal/types").TeardownLogic'.
      Types of parameters 'subscriber' and 'subscriber' are incompatible.
        Type 'import("D:/Projects/opensource/flowaccount-nx-plugins/node_modules/rxjs/internal/Subscriber").Subscriber<any>' is not assignable to type 'import("D:/Projects/opensource/flowaccount-nx-plugins/libs/nx-serverless/node_modules/rxjs/internal/Subscriber").Subscriber<any>'.
          Property 'isStopped' is protected but type 'Subscriber<T>' is not a class derived from 'Subscriber<T>'.

38   const architect = new Architect(architectHost, registry);
                                                    ~~~~~~~~

libs/nx-serverless/src/utils/target.schedulers.ts:21:9 - error TS2345: Argument of type 'MonoTypeOperatorFunction<BuilderOutput>' is not assignable to parameter of type 'OperatorFunction<BuilderOutput, BuilderOutput>'.
  Types of parameters 'source' and 'source' are incompatible.
    Type 'import("D:/Projects/opensource/flowaccount-nx-plugins/node_modules/rxjs/internal/Observable").Observable<import("D:/Projects/opensource/flowaccount-nx-plugins/node_modules/@angular-devkit/architect/src/api").BuilderOutput>' is not assignable to type 'import("D:/Projects/opensource/flowaccount-nx-plugins/libs/nx-serverless/node_modules/rxjs/internal/Observable").Observable<import("D:/Projects/opensource/flowaccount-nx-plugins/node_modules/@angular-devkit/architect/src/api").BuilderOutput>'.
      The types of 'source.operator.call' are incompatible between these types.
        Type '(subscriber: import("D:/Projects/opensource/flowaccount-nx-plugins/node_modules/rxjs/internal/Subscriber").Subscriber<any>, source: any) => import("D:/Projects/opensource/flowaccount-nx-plugins/node_modules/rxjs/internal/types").TeardownLogic' is not assignable to type '(subscriber: import("D:/Projects/opensource/flowaccount-nx-plugins/libs/nx-serverless/node_modules/rxjs/internal/Subscriber").Subscriber<any>, source: any) => import("D:/Projects/opensource/flowaccount-nx-plugins/libs/nx-serverless/node_modules/rxjs/internal/types").TeardownLogic'.
          Types of parameters 'subscriber' and 'subscriber' are incompatible.
            Type 'import("D:/Projects/opensource/flowaccount-nx-plugins/libs/nx-serverless/node_modules/rxjs/internal/Subscriber").Subscriber<any>' is not assignable to type 'import("D:/Projects/opensource/flowaccount-nx-plugins/node_modules/rxjs/internal/Subscriber").Subscriber<any>'.
              Property 'isStopped' is protected but type 'Subscriber<T>' is not a class derived from 'Subscriber<T>'.

21         filter(e => e.success !== undefined),
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
libs/nx-serverless/src/utils/testing.ts:38:50 - error TS2345: Argument of type 'CoreSchemaRegistry' is not assignable to parameter of type 'SchemaRegistry'.
  The types of 'compile(...).source.operator.call' are incompatible between these types.
    Type '(subscriber: import("D:/Projects/opensource/flowaccount-nx-plugins/libs/nx-serverless/node_modules/rxjs/internal/Subscriber").Subscriber<any>, source: any) => import("D:/Projects/opensource/flowaccount-nx-plugins/libs/nx-serverless/node_modules/rxjs/internal/types").TeardownLogic' is not assignable to type '(subscriber: import("D:/Projects/opensource/flowaccount-nx-plugins/node_modules/rxjs/internal/Subscriber").Subscriber<any>, source: any) => import("D:/Projects/opensource/flowaccount-nx-plugins/node_modules/rxjs/internal/types").TeardownLogic'.
      Types of parameters 'subscriber' and 'subscriber' are incompatible.
        Type 'import("D:/Projects/opensource/flowaccount-nx-plugins/node_modules/rxjs/internal/Subscriber").Subscriber<any>' is not assignable to type 'import("D:/Projects/opensource/flowaccount-nx-plugins/libs/nx-serverless/node_modules/rxjs/internal/Subscriber").Subscriber<any>'.
          Property 'isStopped' is protected but type 'Subscriber<T>' is not a class derived from 'Subscriber<T>'.

38   const architect = new Architect(architectHost, registry);
~

**Comment by wickstargazer** [comment]
any idea guys? i will take a hack at it again to try make it build

**Comment by wickstargazer** [comment]
guys i have moved the needle. Its building now but test is not passing, i am figuring out how to run schematic for `jest.conf` its not working the same way :) So it the tests passes and e2e passes then we are good to go!

**Comment by wickstargazer** [comment]
Hi guys, linking the pr to this issue https://github.com/flowaccount/nx-plugins/pull/80

Also published a preview version for the lib. [1.0.0-beta.1](https://www.npmjs.com/package/@flowaccount/nx-serverless/v/1.0.0-beta.1)

**Comment by wickstargazer** [comment]
@jbojcic1 @dbrody @Nightbr i have published `1.0.0-beta.2` the main commands are working now
build, serve, deploy, destroy

have to test out sls and compile. Finish the unit test fix docs and release the 1.0.0!
